              requestId));
    } catch (RuntimeException ex) {
      throw new SystemException(
          ErrorCode.EXTERNAL_SERVICE_ERROR,
          "Unexpected error while uploading CSV to S3",
          ex,
          null,
          Map.of(
              "operation",
              "saveFileToS3",
              "bucket",
              bucket,
              "key",
              s3Path,
              "requestId",
              requestId));
    }
  }
